ccdproc/core.py

```diff
@@ -262,6 +265,10 @@ def ccd_process(ccd, oscan=None, trim=None, error=False, master_bias=None,
         raise TypeError(
             'master_flat is not None or a CCDData object.')
 
+    # apply the gain correction only at the end if gain_corrected is False
+    if isinstance(gain, Quantity) and gain_corrected is False:
```

Comment:
`not gain_corrected` seems better than `gain_corrected is False` and it would prevent accidentally not triggering the branch.